src/vs/platform/dialogs/common/dialogs.ts

```diff
@@ -18,100 +18,171 @@ import { isLinux, isMacintosh, isWindows } from 'vs/base/common/platform';
 import { IProductService } from 'vs/platform/product/common/productService';
 import { deepClone } from 'vs/base/common/objects';
 
-export interface FileFilter {
-	readonly extensions: string[];
-	readonly name: string;
+export interface IDialogArgs {
+	readonly confirmArgs?: IConfirmDialogArgs;
+	readonly inputArgs?: IInputDialogArgs;
+	readonly promptArgs?: IPromptDialogArgs;
```

Comment:
Is this a "one of" situation? Maybe document that, or even enforce it more explicitly:
```
export type IDialogArgs = {confirmArgs: IConfirmDialogArgs, inputArgs?: undefined, promptArgs?: undefined}|
    {readonly confirmArgs?: undefined, readonly inputArgs: IInputDialogArgs, promptArgs?: undefined}|
    {readonly confirmArgs?: undefined, readonly inputArgs?: undefined, promptArgs: IPromptDialogArgs};
```
Maybe the verbose TypeScript is not worth it. Maybe one day they will have a OneOf generic type (a bit like https://dev.to/maxime1992/implement-a-generic-oneof-type-with-typescript-22em).